internal/provider/resource_gitlab_runner.go

```diff
@@ -0,0 +1,257 @@
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_runner`" + ` resource allows registering a runner, either at an instance level
+or at a group level. The runner will be registered at a group level if the token used is from a group, or at an 
+instance level if the token used is for the instance.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner)`,
+
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabRunnerCreate,
+		UpdateContext: resourceGitLabRunnerUpdate,
+		ReadContext:   resourceGitLabRunnerRead,
+		DeleteContext: resourceGitLabRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Description: `The registration token used to register the runner`,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Sensitive:   true,
+			},
+			// Even though the output variable is just called 'token', we need to differentiate between the regitration
+			// and authentication token in terraform.
+			"authentication_token": {
+				Description: `The authentication token used for building a config.toml file. This value is not present when imported.`,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"status": {
+				Description: `The status of runners to show, one of: online and offline. active and paused are also possible values
+				 which were deprecated in GitLab 14.8 and will be removed in GitLab 16.0`,
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Description: `The runner's description'`,
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"paused": {
+				Description: `Whether the runner should ignore new jobs`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(false),
+			},
+			"locked": {
+				Description: `Whether the runner should be locked for current project`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(false),
+			},
+			"run_untagged": {
+				Description: `Whether the runner should handle untagged jobs`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(true),
+			},
+			"tag_list": {
+				Description: `List of runner’s tags`,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"access_level": {
+				Description: fmt.Sprintf(`The access_level of the runner. Valid values are: %s`,
+					renderValueListForDocs(runnerAccessLevelAllowedValues)),
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(runnerAccessLevelAllowedValues, true),
+				Default:      "not_protected",
```

Comment:
I prefer to not duplicate the API defaults in the provider if not really necessary.

Reply:
Same comment as above about the import step of the test. Is there a better way to handle this?

Reply:
Yes, as the other attribute: mark it as `Computed` :)